src/components/Tags/Tag.utils.ts

```diff
@@ -1,42 +1,50 @@
 import { Tag } from "../../features/entity.types";
 
-export const defaultTagsList: string[] = [
+export const DefaultTagsList: string[] = [
```

Comment:
```suggestion
export const DefaultTagsList = [
  "pests",
  "disease",
  "sowed",
  "seedling",
  "default",
] as const;

type DefaultTag = typeof DefaultTagsList[number]

export const convertToTag = (tagName: string) => {
  const tagNameToTagTable: Record<DefaultTag, Tag> = {
    pests: {
      tagLabel: tagName,
      tagColor: "#FF5A33",
      tagIcon: "pest",
    },
    disease: {
      tagLabel: tagName,
      tagColor: "#633c15",
      tagIcon: "disease",
    },
    // other default tags
  }

  return DefaultTagsList.includes(tagName) ? tagNameToTagTable[tagName]: tagNameToTagTable.default
```
you wanna avoid switches if you can, you have no way of typing the cases. you can easily forget a case or have one incorrect. literally nearly had a bug got to production a few weeks ago that I even code reviewed, we realised it was missing a case just before they merged 😅 

TS 5 actually just introduced something to help with switches though it's not actually provide type coverage but auto-filling the cases 
https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/#exhaustive-switch-case-completions

Reply:
Saw this today, looks like you can jank ts to make a switch exhaustive. I reckon a lookup table would be better in most cases but it’s an option. https://stackoverflow.com/a/39419171